### DISCUSSION TITLE
Is it possible to overload the call operator?

### DISCUSSION THREAD

**Discussion by Cazadorro** [shader-slang/slang#4047] [Q&A] [answered]
I see subscript is overloadable, but I don't see anything about the call operator `()`.  There's no negative documentation that this is not available AFAICT, and there's lots of magic methods usable I don't understand beyond __subscript.

**Comment by natevm** [comment]
For construction at least, you’d use __init(params…). That’s not the same as call operator overloading, but figured I’d put it out there for context.

**Reply by csyonghe** [reply]
Overloading for operator() is not supported yet. It is possible for us to add the feature, but we are currently very short on engineering resources.

**Comment by csyonghe** [comment, marked as answer]
This is now supported, closing now.

```
struct MyFunctor
{
       int operator()(float v) { ... }
}

void test()
{
    let f = MyFunctor();
    f(2.0f);
}
```

**Reply by natevm** [reply]
Awesome! Looking forward to using this.

**Reply by Cazadorro** [reply]
Damn you guys are doing overtime on this project! Seriously did not expect this in for a very long time!
